robot_framework/ode_ingest/file_sorting.py

```
@@ -0,0 +1,84 @@
+"""Contains functions for sorting CSV files by date and sequence number, ensuring correct processing."""
+
+import re
+from datetime import datetime
+from pathlib import Path
+from typing import List, Tuple, Optional
+
+
+def _extract_date_from_filename(filename: str) -> Optional[datetime]:
+    """
+    Extract date stamps from filename in format: YYYY-MM-DD
+
+    Args:
+        filename: Filename, eg. '0751_ODE_2025-06-17_002_01-Bilag-master_Delta_001af2.csv'
+
+    Returns:
+        datetime object or None
+    """
+    date_pattern = r'(\d{4}-\d{2}-\d{2})'
+    match = re.search(date_pattern, filename)
+
+    if match:
+        date_str = match.group(1)
+        try:
+            return datetime.strptime(date_str, '%Y-%m-%d')
+        except ValueError:
+            return None
```

Do we want to raise an error instead? Don't we always expect a date?